src/main/java/net/minecraftforge/items/IItemHandler.java

```diff
@@ -61,12 +76,134 @@
     @NotNull
     ItemStack insertItem(int slot, @NotNull ItemStack stack, boolean simulate);
 
+    /**
+     * <p>
+     * Inserts a list of itemstacks into the inventory and return the remainder.
+     * The list of itemstacks and its contents <em>should not</em> be modified in this function!
+     * The distribution of the items is left to the ItemHandler.
+     * </p>
+     * <p>
+     * Even though a default implementation is provided,
+     * if your inventory behaves in "nonstandard" ways, you may have to override it.
+     * </p>
+     * <p>
+     * This method must have the four properties required for {@link IItemHandler#insertItem(int, ItemStack, boolean)}.
+     * </p>
+     *
+     * @param stacks   List of items to insert. Contents must not be null.
+     * @param simulate If true, the insertion is only simulated
+     * @return An array with the same length as the input list representing the remaining itemstacks.
+     *         Each element of the array has the same properties as the return value of
+     *         {@link IItemHandler#insertItem(int, ItemStack, boolean)}
+     *         (non-null, could be the same as its corresponding element in input list, etc.).
+     */
+    @NotNull
+    default ItemStack[] insertItems(@NotNull List<ItemStack> stacks, boolean simulate)
```

Comment:
Why the mixing of lists and arrays?  This should probably be both `List<ItemStack>` or both `ItemStack[]`.

Reply:
There is a method to turn an array into a list but not the other way around, so taking a list but returning array is the most flexible. Also, returning an array signifies it's "fixed size", since in our case each index in the array corresponds to what remains of the index in the list. 

Anyhow, I can totally change it though if you'd like.